ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/FileObjectGetter.java

```diff
@@ -47,14 +52,27 @@ public SeekableByteChannel buildChannel(final String key) throws IOException {
             Files.createDirectories(FileUtils.resolveForSymbolic(parentPath));
         }
 
-        if ( ! FileUtils.isTransferablePath(objectPath)) {
+        if (!FileUtils.isTransferablePath(objectPath)) {
             throw new UnrecoverableIOException(objectPath + " is not a regular file.");
         }
 
+        try {
+            striped.get(key).lock();
```

Comment:
Why not get the lock once instead of twice?

Reply:
This is fixed, we only call striped.get once now. See changes in 878c16a